### ISSUE TITLE
Other Features

### ISSUE THREAD

**Issue by rednuz** [BoostIO/Boostnote#94]
First of all, congratulations. This project is fantastic.

I have been looking exactly for this for over a week; A blend between Atom, Quiver and Evernote. Simply perfect. Seriously, I am very, very impressed and excited for Boostnote.

I thought I would add some keypoints that would make this little app the ultimate writing tool:
- Menubar quick note scratchpad (see MiniNote Pro [menubar icon needs to be black and white btw]).
- Word and character counter.
- Extensive theme styling and customization (I am using OS X and its very limited).
- Multiple note selection for deletion (I would personally get rid of the blue border and just add a bit of a darker background for selection: http://i.share.pho.to/4bca6da7_o.png).
- Proper dropbox integration with encryption (see: https://laverna.cc/).
- Each storage saves to separate folder. Each snippet saves in a subfolder with the according file syntax extension e.g.: MyStorage>Default(4bc6eb15d8b8)>MySnippet>TabOneFilename.html
- Import folders to left sidebar (dropdown tree structure).
- Distraction free and dual panel live preview modes.

That's all I can come up with for now. I would love to donate for this project too if you have a link you should add it to the website.

**Comment by Rokt33r** [comment]
I'm very very excited your feedback.

Btw, these are my answer to each item.

> Menubar quick note scratchpad

Check #85. There are some comment about this. We can talk more about this issue.
And I agree with that the icon should be black & white.

> Word and character counter

Should I count all of characters(including markdown syntax) or only rendered text? 

> Extensive theme styling and customization

I was going to implement it on the last update, v0.6.3. I'd tried to set the colors to inline style instead of CSS. But, I noticed that, like `:hover`, some selector isn't available on inline style.
So, this is currently pending now. 
But, It doesn't mean I gave up. I think it can be done by re-compiling customized style and saving it to some kind of cache(localStorage would be fine). So, It would be a matter of time.

> Multiple note selection for deletion

It should be. Btw, I want to ask something. Is there any other function for multiple note selection?
I think moving notes to other storage or folders might be a valid choice.

FYI, OS X Notes only provides these options.

<img width="123" alt="screen shot 2016-08-25 at 10 12 01 am" src="https://cloud.githubusercontent.com/assets/5865853/17953212/756219b0-6aac-11e6-8910-212ec6eef8dc.png">


> (I would personally get rid of the blue border and just add a bit of a darker background for selection: http://i.share.pho.to/4bca6da7_o.png)

This can be solved after the customization is available.

> Proper dropbox integration with encryption (see: https://laverna.cc/).
Cool. I'll check it. We should talk about more. If you have an detailed idea about this, make a new issue like #80 and we could be implement this much faster.

>  Each storage saves to separate folder. Each snippet saves in a subfolder with the according file syntax extension e.g.: MyStorage>Default(4bc6eb15d8b8)>MySnippet>TabOneFilename.html

Hmm... I can't sure this should be implemented. By this, users can edited their note from their own editor. But, It makes hard to maintain the storage.
In fact, I was going to collapse all of folders and put all of their notes into a single folder. #89 
I think we should talk about this more.

> Import folders to left sidebar (dropdown tree structure).

I can't understand what it means. Could you explain more about this?

> Distraction free and dual panel live preview modes.

It should be. I'll ship it soon.

**Comment by brpaz** [comment]
regarding multiple notes selection, I think we should be able to:

* Delete notes
* Move notes to a different Storage
* Change tags

Similar with Evernote.

What would be also nice was the possibility to change order of storage / folders in the sidebar and also possibility to sort notes Alphabetically.

**Comment by twhiting** [comment]
I use boostnote extensively between school and work for my dev notes.  Thank you for creating something so useful, I only wish I knew more about java script / electron so I could contribute effectively. :+1: 

A couple of requests from a daily user:
* An easy way to bold, strike-through, italicize, etc.  Just some formatting buttons where I can highlight some text and wrap it in a code block, bold, etc. by clicking a button would be awesome (just like how github does it in the box i'm writing this post in!
* color text
* Image support (probably on your to-do but not super important), working as a dev inserting images in my dev notes can frequently make taking notes much easier / cleaner.
* Custom themes (would be nice to have a setting to override the .css or something, not critical)
* Scroll past the bottom most line of the editor.  When i'm taking long notes I often have to insert ~20 spaces so that I'm not typing at the very bottom of the editor.
* Full-screen mode (not critical, but nice)
* Full google drive, onedrive support (You seem to be working on this).  I sync my notes between my computers by utilizing google drive / onedrive syncing to local storage.  It'd be nice to utilize their API's so I don't have to manually set up folder syncing (not crucial)

Also here are some features that may or may not make the cut depending on where you want to go with this project, aimed more at enterprise level.
* Authentication / Password lock storage & encryption.  I have dev notes that i'd like to keep encrypted and be unlock-able, this way I can store it on my personal storage and be sure that this data is safe.  
* Multiple users editing notes, so me and someone on my team can edit notes and have them synced between us via onedrive/dropbox/gdrive, etc.  And also a last edited by _username_ message.  No need to setup an account or anything, rather just have a nickname or something for basic identification.
* Save via git repository.  This would be a pretty neat feature.  Instead of utilizing dropbox/onedrive/gDrive push changes to a local/remote git repository. 

Again, these are all just ideas for the project and it's completely up to you if/which you implement.

**Comment by Rokt33r** [comment]
@twhiting Thanks for your feedback! 

>An easy way to bold, strike-through, italicize, etc. Just some formatting buttons...

It should be and not so difficult. I think I can ship it on this week.

> color text

It is possible. Boostnote try to comply Common Markdown. All html tag will be rendered just as it.(To prevent XSS, we use sandbox iframe as a renderer.)

> Image support

Check #93 . Although it seems to be inactive now, I consider it is a quite important feature.
I'll work it, after #100 done.

> Full-screen mode

I need more explain. There are so many options related in fullscreen. 
Ex) Fullscreen whole app, only preview, 2-pane(editor and preview)

> Scroll past the bottom most line of the editor.

I'll try to fix it on this week. But, It might be shipped on the next week.

> Security and Cloud issues

I gladly want to do. Just the only problem is my lack of skill and time.
Currently, I'd been developing dataApi and its test. With this data api, I think Boostnote storage can be served by any NodeJS server. After refactoring dataApi done, I'll start integrating each cloud service.
About security, I think it quite takes lots of time and more information. We have to discuss this issue more.

Btw, We have a slack. if you want to join us, send me an e-mail. fluke8259@gmail.com

**Comment by kirtan403** [comment]
> It should be and not so difficult. I think I can ship it on this week.

Still no shortcuts for markdown bold?

**Comment by asmsuechan** [comment]
Hi, @kirtan403! I removed it (https://github.com/BoostIO/Boostnote/pull/399) because it had a conflict.

**Comment by kirtan403** [comment]
@asmsuechan Ohh, Are there any other shortcuts for writing into markdown?

**Comment by asmsuechan** [comment]
Here's a list of shortcuts https://github.com/BoostIO/Boostnote/issues/533#issuecomment-298424118

**Comment by kirtan403** [comment]
Thanks for the update. But I see nothing for the writing in markdown. All the shortcuts are for the navigation/actions of the app. And as the issue suggest, putting it somewhere in the wiki or documentation? That would be great.

**Comment by alexanderadam** [comment]
I'm not sure whether an issue actually containing more than one issue is a good idea. Especially for tracking the progress.

I guess for tracking the encryption plus sync issue it would be better to use a dedicated issue like one of these: #33, #827 or #1511
